test/micro/org/openjdk/bench/java/security/PKCS12KeyStores.java

```diff
@@ -0,0 +1,199 @@
+            byte[] x1 = xeh("3041020100301306072A8648CE3D020106082A8648CE3D03" +
+                    "0107042730250201010420B561D1FBE150488508BBE8FF4540F09057" +
+                    "58712F5D2D3CC80F9A15BA5D481117");
+            byte[] x2 = xeh("3082012D3081D5A00302010202084EE6ECC5585640A7300A" +
+                    "06082A8648CE3D040302300C310A30080603550403130161301E170D" +
+                    "3230313131373230343730355A170D3233303831343230343730355A" +
+                    "300C310A300806035504031301613059301306072A8648CE3D020106" +
+                    "082A8648CE3D030107034200041E761F511841602E272B40A021995D" +
+                    "1BD828DDC7F71412D6A66CC0CB858C856D32C58273E494676D1D2B05" +
+                    "B8E9B08207A122265C2AA5FCBDCE19E5E88CA7A1B6A321301F301D06" +
+                    "03551D0E04160414173F278D77096E5C8EA182D12F147694587B5D9A" +
+                    "300A06082A8648CE3D04030203470030440220760CEAF1FA7041CB8C" +
+                    "1CA80AF60E4F9C9D5136D96B2AF0AAA9440F79561C44E502205D5C72" +
+                    "886C92B95A681C4393C67AAEC8DA9FD7910FF9BF2BCB721AE71D1B6F88");
+            KeyFactory kf = KeyFactory.getInstance("EC");
+            pk = kf.generatePrivate(new PKCS8EncodedKeySpec(x1));
+            CertificateFactory cf = CertificateFactory.getInstance("X.509");
+            certs = new Certificate[] { cf.generateCertificate(new ByteArrayInputStream(x2)) };
+
+            bw2048 = genweak2048();
+            bw50000 = genweak50000();
+            bs50000 = genstrong50000();
+            bs10000 = genstrong10000();
+            bs2048 = genstrong2048();
+        } catch (Exception e) {
+            throw new AssertionError(e);
+        }
+    }
+
+    @Benchmark
+    public KeyStore inweak2048() throws Exception {
+        return in(bw2048);
+    }
+    @Benchmark
+    public KeyStore inweak50000() throws Exception {
+        return in(bw50000);
+    }
+    @Benchmark
+    public KeyStore instrong50000() throws Exception {
+        return in(bs50000);
+    }
+    @Benchmark
+    public KeyStore instrong10000() throws Exception {
+        return in(bs10000);
+    }
+    @Benchmark
+    public KeyStore instrong2048() throws Exception {
+        return in(bs2048);
+    }
+
+    static KeyStore in(byte[] b) throws Exception {
+        KeyStore ks = KeyStore.getInstance("pkcs12");
+        ks.load(new ByteArrayInputStream(b), pass);
+        if (!ks.getCertificate("a").getPublicKey().getAlgorithm().equals(
+                ks.getKey("a", pass).getAlgorithm())) {
+            throw new RuntimeException("Not same alg");
+        }
+        return ks;
+    }
+
+    @Benchmark
+    public byte[] outweak2048() throws Exception {
+        return genweak2048();
+    }
+
+    private static byte[] genweak2048() throws Exception {
+        return out("PBEWithSHA1AndRC2_40", "2048",
+                "PBEWithSHA1AndDESede", "2048",
+                "HmacPBESHA1", "2048");
+    }
```

Comment:
Any reason why these methods are not inlined into `@Benchmark` methods right away? E.g. why it is not:

```
    @Benchmark
    public byte[] outweak2048() throws Exception {
        return out("PBEWithSHA1AndRC2_40", "2048",
                "PBEWithSHA1AndDESede", "2048",
                "HmacPBESHA1", "2048");
    }
```

Reply:
I want to make the byte[] filed static so the out() and gen***() methods must be static, and there is a warning on using static method as a `@Benchmark` method.